component/pyroscope/ebpf/ebpfspy/sd/procfs.go

```diff
@@ -0,0 +1,37 @@
+package sd
+
+import (
+	"bufio"
+	"fmt"
+	"regexp"
+)
+
+var (
+	cgroupContainerIDRe = regexp.MustCompile(`^.*/(?:.*-)?([0-9a-f]+)(?:\.|\s*$)`)
+)
+
+func (tf *TargetFinder) getContainerIDFromPID(pid uint32) containerID {
+	f, err := tf.fs.Open(fmt.Sprintf("proc/%d/cgroup", pid))
+	if err != nil {
+		return ""
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
```

Comment:
I think Scanner has a method `Bytes()` and you can run the regex on it. This way you can avoid an allocation for each line.